pkg/ccm/loadbalancer.go

```diff
@@ -163,9 +164,7 @@ func (lb *LBManager) getServicePortMap(service *v1.Service) (map[string]int32, m
 	for _, port := range service.Spec.Ports {
 		typeToInternalPort[strings.ToLower(port.Name)] = port.NodePort
 		typeToExternalPort[strings.ToLower(port.Name)] = port.Port
-		if port.AppProtocol != nil {
-			nameToProtocol[strings.ToLower(port.Name)] = strings.ToUpper(*port.AppProtocol)
-		}
+		nameToProtocol[strings.ToLower(port.Name)] = strings.ToUpper(string(port.Protocol))
```

Comment:
Can we do this in an `else`. For example,
```
if port.AppProtocol != nil {
    nameToProtocol[strings.ToLower(port.Name)] = strings.ToUpper(*port.AppProtocol)
} else {
    nameToProtocol[strings.ToLower(port.Name)] = strings.ToUpper(string(port.Protocol))
}
```

Reply:
> I think this would be a good option to make it possible to use the `appProtocol` here. However, I have a concern. Since we initially use the `protocol` field [on creation](https://github.com/vmware/cloud-provider-for-cloud-director/blob/2d6e9efaf23df037d4cb1d58484bf04b3f2b8935/pkg/vcdsdk/gateway.go#L1754-L1755) it would still mean to alter the monitor on subsequent upgrades. In this case we should likely prefer the `appProtocol` also on creation.


This is a good point. I am not sure of what the preference is according to Kubernetes principles. Since the `protocol` is always present and the `appProtocol` is optional, I think it is logical to assume that the`appProtocol` overrides the `protocol` field.

We should use both the `protocol` and `appProtocol` in the other cases as well. I think it could be handled as another PR, if you want to merge this earlier.

Reply:
Sure, I'm okay with that.
In this case do we want to accept other protocols than `tcp` ? As of right now that is the only health monitor type that will be created as per https://github.com/vmware/cloud-provider-for-cloud-director/blob/2d6e9efaf23df037d4cb1d58484bf04b3f2b8935/pkg/vcdsdk/gateway.go#L751-L752

Reply:
I think we should do it. When I wrote that part of code, I think that there was only `TCP` or I may have made a mistake :(  It deserves another ticket / contribution.

Reply:
Apologies for the delay.
For the time being I have applied the suggestion you made (using an if-else statement) as an intermediate fix.
The larger change regarding other protocols and changing the creation logic as well could then be made in a separate PR.